File: tools/buildReferenceFolder.py
import os
import json
import sys


def build(targetPack):
    packPath = os.path.join(".\src", "packs", targetPack)
    packList = os.listdir(packPath)

    # folderStruct = {"_id": {"name": "foobar", "parent": "_id"}}
    folderStruct = {}

    for p in packList:
        # print(p)
        with open(os.path.join(packPath,p), "r") as read_file:
            data = json.load(read_file)
            if 'type' not in data:
                continue
            if (data["type"] == "Item"):
                folderStruct[data["_id"]] = {"name": data["name"], "parent": data["folder"]}

    with open(os.path.join(".", "tools", "referenceFolders", targetPack + ".json"), "w") as write_file:
        write_file.write(json.dumps(folderStruct, indent=2))

if len(sys.argv) > 1:
    t = sys.argv[1]
    build(t)

File: tools/buildReferenceFolder.mjs
import { promises as fs } from "fs";
import yaml from "js-yaml";
import path from "path";

export default async function buildReferenceFolder(pack) {
  // folderStruct = {"_id": {"name": "foobar", "parent": "_id"}}
  const folderStruct = {};
  const packList = await fs.readdir(path.join("src", "packs", pack));
  for (const p of packList) {
    const file = await fs.readFile(path.join("src", "packs", pack, p), "utf8");
    const data = yaml.load(file);
    if (!data.hasOwnProperty("type")) continue;
    folderStruct[data._id] = {
      name: data.name,
      parent: data.folder,
    };
  }
  await fs.writeFile(
    path.join("tools", "referenceFolders", pack + ".json"),
    yaml.dump(folderStruct, null, 2),
    "utf-8"
  );
}
